src/Entities/ArrowEntity.cpp

```diff
@@ -97,6 +98,9 @@ void cArrowEntity::OnHitSolidBlock(Vector3d a_HitPos, eBlockFace a_HitFace)
 	// Broadcast arrow hit sound
 	m_World->BroadcastSoundEffect("entity.arrow.hit", m_HitBlockPos, 0.5f, static_cast<float>(0.75 + (static_cast<float>((GetUniqueID() * 23) % 32)) / 64));
 
+	// To make buttons get pressed if they got hit by the arrow
+	cBlockButtonHandler::OnArrowHit(*m_World, m_HitBlockPos, a_HitFace);
```

Comment:
The comment reads weird.

Comments are there to make the code more readable, not to let out your creative writing spirit. Although we enjoy the joke from time to time, readability is the primary objective.